src/libraries/Common/src/Interop/Browser/Interop.Runtime.Bridge.cs

```diff
@@ -0,0 +1,369 @@
+                    {
+                        _boundObjects.Remove(jsId);
+                    }
+                    if (obj != null)
+                    {
+                        obj.JSHandle = -1;
+                        obj.IsDisposed = true;
+                        obj.RawObject = null;
+                        obj.Handle.Free();
+                    }
+                }
+            }
+        }
+
+        internal static void UnBindRawJSObjectAndFree(int gcHandle)
+        {
+            GCHandle h = (GCHandle)(IntPtr)gcHandle;
+            JSObject? obj = h.Target as JSObject;
+            lock (_rawToJS)
+            {
+                if (obj?.RawObject != null)
+                {
+                    _rawToJS.Remove(obj.RawObject);
+
+                    int exception;
+                    ReleaseHandle(obj.JSHandle, out exception);
+                    if (exception != 0)
+                        throw new JSException($"Error releasing handle on (js-obj js '{obj.JSHandle}' mono '{(IntPtr)obj.Handle} raw '{obj.RawObject != null})");
+
+                    // Calling Release Handle above only removes the reference from the JavaScript side but does not
+                    // release the bridged JSObject associated with the raw object so we have to do that ourselves.
+                    obj.JSHandle = -1;
+                    obj.IsDisposed = true;
+                    obj.RawObject = null;
+
+                    obj.Handle.Free();
+                }
+            }
+        }
+
+        internal static object CreateTaskSource(int jsId)
+        {
+            return new TaskCompletionSource<object>();
+        }
+
+        internal static void SetTaskSourceResult(TaskCompletionSource<object> tcs, object result)
+        {
+            tcs.SetResult(result);
+        }
+
+        internal static void SetTaskSourceFailure(TaskCompletionSource<object> tcs, string reason)
+        {
+            tcs.SetException(new JSException(reason));
+        }
+
+        internal static int GetTaskAndBind(TaskCompletionSource<object> tcs, int jsId)
+        {
+            return BindExistingObject(tcs.Task, jsId);
+        }
+
+        internal static int BindExistingObject(object rawObj, int jsId)
+        {
+            JSObject? obj = rawObj as JSObject;
+            lock (_rawToJS)
+            {
+                if (obj == null && !_rawToJS.TryGetValue(rawObj, out obj))
+                    _rawToJS[rawObj] = obj = new JSObject(jsId, rawObj);
+
+                return obj == null ? 0 : (int)(IntPtr)obj.Handle;
+            }
+        }
+
+        internal static int GetJSObjectId(object rawObj)
+        {
+            JSObject? obj = rawObj as JSObject;
+            lock (_rawToJS)
+            {
+                if (obj == null && !_rawToJS.TryGetValue(rawObj, out obj))
+                    return -1;
+
+                return obj?.JSHandle ?? -1;
+            }
+        }
+
+        internal static object? GetMonoObject(int gcHandle)
```

Comment:
"GetMonoObject"... should such references to Mono be changed to Net or DotNet or Something like that?

Reply:
Addressed in recent commits